minio/minio
# [10-28.sets branch] Use refresh locking mechanism in locks

## Description
Backporting refresh locking mechanism in 10-28 branch

## Motivation and Context
Backporting refresh locking mechanism in 10-28 branch


## How to test this PR?
- All type of S3 operations, put/get/copy, small and large files and see if it breaks

## Types of changes

## Checklist:
